build/storage/python_system_tools/test_platform.py

```diff
@@ -0,0 +1,262 @@
+                f"Installing {program} failed with code {rc} and output {out}"
+            )
+            return False
+        logging.info(f"{program} installed successfully or present on a machine")
+        return True
+
+    def setup_docker_compose(self) -> bool:
+        """
+        Checks if docker-compose is installed and installs it if it's not
+
+        Returns
+        -------
+        bool
+            True if setup is successful or docker-compose is already installed else False
+        """
+
+        cmds = (
+            'curl -L "https://github.com/docker/compose/releases/download/1.29.2/docker-compose-$(uname -s)-$(uname -m)" -o /usr/local/bin/docker-compose',
+            "chmod +x /usr/local/bin/docker-compose",
+            "ln -s /usr/local/bin/docker-compose /usr/bin/docker-compose",
+        )
+        _, rc = self.terminal.execute("docker-compose --version", raise_on_error=False)
+        if rc:
+            logging.info("Setting up docker-compose")
+            for cmd in cmds:
+                out, rc = self.terminal.execute_as_root(cmd)
+                if rc:
+                    logging.error(
+                        f"Setting up docker failed with code {rc} and output {out}"
+                    )
+                    return False
+            logging.info(f"Setup of docker-compose was successful")
+            return True
+        logging.info(f"docker-compose is already installed")
+        return True
+
+    def setup_libguestfs_tools(self) -> bool:
+        """
+        Installs libguestfs-tools for a specific OS
+
+        Returns
+        -------
+        bool
+            True if installation is successful else False
+        """
+
+        program = "libguestfs-tools"
+        if self.os == "Fedora":
+            program += "-c"
+        return self.install(program)
+
+    def change_vmlinuz(self) -> bool:
+        """
+        Changes the mode of /boot/vmlinuz-*
+
+        Returns
+        -------
+        bool
+            True if change is successful else False
+        """
+
+        out, rc = self.terminal.execute_as_root("chmod +r /boot/vmlinuz-*")
+        if rc:
+            logging.error(
+                f"Changing the mode of /boot/vmlinuz-* failed with code {rc} and output {out}"
+            )
+            return False
+        logging.info("Mode of /boot/vmlinuz-* was changed")
+        return True
+
+    def check_security_policies(self) -> bool:
+        """
+        Checks if security policies are disabled
+
+        Returns
+        -------
+        bool
+            True if security policies are disabled else False
+        """
+
+        if self.os == "Fedora":
+            cmd = "setenforce 0"
+        else:
+            cmd = "systemctl stop apparmor"
+        out, rc = self.terminal.execute_as_root(cmd, raise_on_error=False)
+        if "disabled" in out:
+            logging.info("Security policies are disabled")
+            return True
+        logging.error(
```

Comment:
like above check should log error only if checking operatiion failed

Reply:
changed to info